 ## Mean and Standard Deviation of Barcode Error Part 2

The formula for calculating the mean is:

$$ \bar{X} = \frac{\sum_{1}^{n}{x_n}}{n} $$ 

The formula for calculating the standard deviation is:

$$ \sigma = \sqrt{\frac{(X - \bar{X})^2}{n}} $$ 

In [3]:
barcodeErrors = [36,14,21,39,11,2, 33, 45, 34, 17, 1, 29]

meanErrors = sum(barcodeErrors) / len(barcodeErrors)

$$ \bar{X_2} = \frac{36+14+21+39+11+2 + 33 + 45 + 34 + 17 + 1 + 29}{12} $$ 

In [4]:
print(meanErrors)

23.5


$$ \bar{X_2} = 23.5 $$ 

The new mean differs from the former mean as follows:

$$ \bar{X_2} - \bar{X_1} = 23.5 - 20.5 = 3 $$ 

In [5]:
for i in barcodeErrors:
    print((i-meanErrors)**2)

156.25
90.25
6.25
240.25
156.25
462.25
90.25
462.25
110.25
42.25
506.25
30.25


$$ \sigma_2 = \sqrt{\frac{(36-23.5)^2 + (14 - 23.5)^2 + (21 - 23.5)^2 + (39 - 23.5)^2 + (11 - 23.5)^2 + (2 - 23.5)^2 + (33 - 23.5)^2 + (45 - 23.5)^2 + (34 - 23.5)^2 + (17 - 23.5)^2 + (1 - 23.5)^2 + (29 - 23.5)^2}{12}} $$ 

$$ \sigma_2 = \sqrt{\frac{156.25 + 90.25 + 6.25 + 240.25 + 156.25 + 462.25 + 90.25 + 462.25 + 110.25 + 42.25 + 506.25 + 30.25}{12}} $$ 

In [9]:
sqDif = [156.25 + 90.25 + 6.25 + 240.25 + 156.25 + 462.25 + 90.25 + 462.25 + 110.25 + 42.25 + 506.25 + 30.25]
print(sum(sqDif))

2353.0


$$ \sigma_2 = \sqrt{\frac{2353}{12}} $$ 

In [10]:
2353 / 12 

196.08333333333334

$$ \sigma_2 = \sqrt{196.083333} $$ 

In [11]:
import math as m

print(m.sqrt(196.08333333))

14.002975874077624


$$ \sigma \approx 13.276 $$ 